pkg/vtpm/src/server.cpp

```diff
@@ -340,13 +352,19 @@ parseRequest(int sock,
 int
 handleRequest (int sock, google::protobuf::uint32 size)
 {
-    char payload[size+CODED_STRM_HDR_LEN];
     eve_tools::EveTPMRequest request;
     eve_tools::EveTPMResponse response;
     int byteCnt = 0, rc = success;
-    const char *cmdArgs[MAX_ARGS+1];
+    const char *cmdArgs[MAX_ARGS+1] = {};
     ifstream cmdOut;
 
+    char *payload = new(nothrow) char [size+CODED_STRM_HDR_LEN];
+    if (payload == NULL) {
+        cerr << "Failed to allocate memory for requests payload" << std::endl;
+        rc = failure;
+        goto cleanup_and_exit;
```

Comment:
In case if it's not expensive and possible for us to use exceptions here I'd put one handle on highest level function to cleanup and exit so that we can throw and forget

Reply:
this code is a mix of c and c++ and this pattern is used socket APIs which won't throw exception.

Reply:
I would decompose it then to functions and check error returning from each, but refactoring and cleaning up is not is not part of PR